functions/callback-functions.md

```diff
@@ -0,0 +1,214 @@
+
+For example, let's imagine this problem and pseudocode:
+
+We need to build one calculator function that can add, subtract, multiply, and divide two numbers.
+
+In order to accomplish that goal, we need three arguments:
+
+1. The first number
+2. The second number
+3. Something that represents to operation we need, to help us determine if we're adding, subtracting, multiplying, or dividing
+
+We could approach the third argument by expecting a string, such as `'add'`, `'+'`, `'multiply'`, or `'*'`.
+
+<!-- prettier-ignore-start -->
+```js
+const calculator = function(a, b, operation){
+  // if (operation === 'add') {
+  //  return a + b;
+  // }
+}
+```
+<!-- prettier-ignore-end -->
+
+However, wouldn't it be grand if we could _pass in_ the actual operation, whether it's addition or subtraction, and execute it?
+
+## Callback Functions
+
+In order to pass around behavior, we pass around functions.
+
+More specifically, a function can _take in a function_ as an _argument_, and then call that function.
+
+To help us talk about functions, we can use specific vocabulary around these functions. A **callback function** is a nickname for a function that is _passed into_ another function.
+
+We can reverse that definition and imagine a function that _takes in another function as an argument_. The function passed in as an argument can be labeled as a _callback function_.
+
+### Passing Around Cleaning Up Behavior
+
+Let's dive into an example! Let's define these two functions, `cleanKitchen` and `bakeCake`.
+
+<!-- prettier-ignore-start -->
+```js
+const cleanCountertop = function () {
+  console.log('I\'m cleanin\' my counters now! Scrubba scrub scrub.');
+}
+
+const bakeCake = function () {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+}
+
+bakeCake();
+```
+<!-- prettier-ignore-end -->
+
+The variable `cleanCountertop` references a function that prints details about cleaning a kitchen.
+
+The variable `bakeCake` references a different function that prints details about baking a cake.
+
+When we bake a cake, maybe there's always cleanup behavior that we want to call, whether it's cleaning our counters, sink, or stove-top.
+
+We can _change the function signature_ of `bakeCake` to accept an argument, `cleanupBehavior`. Inside our `bakeCake` function, we can take that argument and _invoke_ whatever function it references, by attaching `()` to it.
+
+<!-- prettier-ignore-start -->
+```js
+const bakeCake = function (cleanupBehavior) {
+  console.log('Make the batter...');
+  console.log('Put it in the oven...');
+  cleanupBehavior();
+}
+```
+<!-- prettier-ignore-end -->
+
+We can refer to `cleanupBehavior` as a _callback function_ inside our `bakeCake` function. After we look at the big picture, we can think about the phrase "callback."
```

Comment:
It seems unnecessary to put such a big focus on the little bit of discussion remaining about the term callback. As long as we are following a natural narrative flow, I think it's fine to get there when we get there.
```suggestion
We can refer to `cleanupBehavior` as a _callback function_ inside our `bakeCake` function.
```